node_pool_taint/main.tf

```diff
@@ -0,0 +1,58 @@
+# IMPORTANT
+## Please BE CAREFUL when changing `keepers`. If you add a keeper
+## then this will cause a CREATION CASCADE! Since a new keeper
+## triggers a new `random_id` resource to generate then it will change
+## the name of the node pool, which will recreate that resource!
+## This is the preferred method for random naming now that
+## name-prefix is being deprecated!
+## IF YOU ADD A KEEPER YOU WILL NEED TO WARN DOWNSTREAM USERS THAT
+## IT WILL TRIGGER A CASCADING NODEPOOL REPLACEMENT.
+## The replacement should be safe because of the create_before_destroy
+## but will cause some questions and confusion.
+## All the current keepers are bound to params that will already cause
+## a -/+ operation for the nodepool.
+# https://github.com/terraform-providers/terraform-provider-google/issues/1054
+resource "random_id" "entropy" {
+  keepers {
+    machine_type = "${var.machine_type}"
+    name         = "${var.name}"
+    region       = "${var.region}"
+    disk_size    = "${var.disk_size_in_gb}"
```

Comment:
Does changing taints recreate the node-pool? If so then it should probably live as a keeper in the entropy for naming resources (due to `create_before_destroy = true`). Same question goes for `labels`. 

Do we also want to maybe join these changes with the other PR I have in flight so we can version bump the node-pool once (to v2.0.0). As it stands right now, changes to the `random_id.entropy` will always recreate the asset when you add a field to the `keeper`. It's an unfortunate byproduct of the `create_before_destroy = true` and keeping random naming.

Reply:
These changes do recreate the node-pool. I added them to the keepers. 

I agree we should merge both of these PRs to v2.0.0.